utils.go

```diff
@@ -214,17 +170,9 @@ func formatError(err error) error {
 	}
 }
 
-func newS3Service(sess *session.Session, cfgs ...*aws.Config) (srv *s3.S3) {
-	srv = s3.New(sess, cfgs...)
-
-	// S3 will calculate payload's content-sha256 by default, we change this behavior for following reasons:
-	// - To support uploading content without seek support: stdin, bytes.Reader
-	// - To allow user decide when to calculate the hash, especially for big files
-	srv.Handlers.Sign.SwapNamed(v4.BuildNamedHandler(v4.SignRequestHandler.Name, func(s *v4.Signer) {
-		s.DisableURIPathEscaping = true
-		// With UnsignedPayload set to true, signer will set "X-Amz-Content-Sha256" to "UNSIGNED-PAYLOAD"
-		s.UnsignedPayload = true
-	}))
+func newS3Service(cfgs *aws.Config) (srv *s3.Client) {
+	srv = s3.NewFromConfig(*cfgs, func(o *s3.Options) {})
```

Comment:
Maybe we should init like this:
```go
client := s3.NewFromConfig(cfg, func(options *s3.Options) {
		options.Region = opt.Location
		if opt.HasForcePathStyle {
			options.UsePathStyle = opt.ForcePathStyle
		}
		// ...
		options.EndpointResolver = s3.EndpointResolverFunc(func(region string, options s3.EndpointResolverOptions) (e aws.Endpoint, err error) {
			e.URL = url
			e.SigningRegion = opt.Location
			return e, err
		})

		options.APIOptions = append(options.APIOptions,
			func(stack *middleware.Stack) error {
				v4.RemoveContentSHA256HeaderMiddleware(stack)
				v4.RemoveComputePayloadSHA256Middleware(stack)
				return v4.AddUnsignedPayloadMiddleware(stack)
			})

		//options.ClientLogMode = aws.LogSigning | aws.LogRequest | aws.LogResponseWithBody
	})
```

Reply:
I also find a way like:
`			loadOptions := []func(*LoadOptions) error{
				WithEndpointResolver(endpointResolver),
				WithAPIOptions([]func(*middleware.Stack) error{
					func(stack *middleware.Stack) error {
						return stack.Initialize.Add(middleware.InitializeMiddlewareFunc("GetRoleArns", func(ctx context.Context, in middleware.InitializeInput, next middleware.InitializeHandler,
						) (
							out middleware.InitializeOutput, metadata middleware.Metadata, err error,
						) {
							switch v := in.Parameters.(type) {
							case *sts.AssumeRoleInput:
								credChain = append(credChain, *v.RoleArn)
							}

							return next.HandleInitialize(ctx, in)
						}), middleware.After)
					},
				}),
			}`
Maybe it's  available too?